DeleteOccurrencesOfExtraElementsInAnArray.rb

```diff
@@ -0,0 +1,57 @@
+=begin
+Do this without using the .uniq method
+
+Create a function that takes two arguments: an array arr and a number num. If an element occurs in arr more than num times, remove the extra occurrence(s) and return the result.
+
+Examples
+
+delete_occurrences([1, 1, 1, 1], 2) ➞ [1, 1]
+
+delete_occurrences([13, true, 13, nil], 1) ➞ [13, true, nil]
+
+delete_occurrences([true, true, true], 3) ➞ [true, true, true]
+Notes
+Do not alter the order of the original array.
+Algorithm :
+
+1.We have a set of elements and a number(occurrence limit ) . we need to find the each element's occurrence in that set which should not be more than given number. The extra occurrence than the given specified occurrence limit should be removed.
+2.Lets take the first element, and compare it with rest all elements on the right to check it's occurrence. By doing this we get the occurrence of the first element. now we should only consider the occurrence up to the given occurrence limit .
+3.Example limit given is 2 and we should only consider 2 occurrence of each element.
+4.Similarly do it for all the elements.
+5 .we should display all the element's occurrences up to the given limit.
+
+every time you do count, you are going through the whole array and checking the number of times this occurs.
+
+try a better algorithm where you only have to go through the array one and then get the result in the end.
+
+[13, true, 13, nil]
+
+everytime you call .count it goes over it
+=end
+
+def delete_occurrences(array, limit)
+  i = 0
+  check = array[i]
+  occurrence = []
+  counter  = 0
+  array.each_with_index do |element, index|
```

Comment:
hmm... im not sure if this will work... can you try this example:

`([1,3,1,3,7,4,1,2,1,4,3,3], 2) -> [1,3,1,3,7,4,2,4]`

So we want to keep a maximum of 2 occurrences of the same number. That is what the second param is saying. 

But for the same array, if the second param to the method is `3`, then the result will look like this for the same array as above:

```
[1,3,1,3,7,4,1,2,4,3,2]
```

Reply:
I have  solved this problem now ! its not the efficient way  but it does work according the problems asked